src/components/editor/markdown-renderer/replace-components/image/image-frame.tsx

```diff
@@ -1,7 +1,25 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { getProxiedUrl } from '../../../../../api/imageProxy'
+import { ApplicationState } from '../../../../../redux'
+
+export const ImageFrame: React.FC<React.ImgHTMLAttributes<HTMLImageElement>> = ({ alt, src, ...props }) => {
+  const [imageUrl, setImageUrl] = useState(src)
```

Comment:
You're using the original src as initial value. That means, that before you get the proxied url, you use the orignal one.

Reply:
Yes, that's a kind of fallback to show an image in any case. It's also the default value for the state as the url-replace-logic won't be triggered at all if the imageProxy setting is not set to true.

Reply:
Yes, but your first rendering would ALWAYS set the original src, then the effect would trigger and THEN after all the ajax and callbacks the proxied image would be shown

OR 

The image proxy is useless because the client is loading the original image :sweat_smile:

Reply:
I also would NEVER use the original one as fallback, because that could leak data. But we could add a boolean to the backendConfig like "fallbackToOriginalImageUrl". Some people prefer availablity and some privacy.

Reply:
You're right, the fallback makes an image proxy useless. I'll change that.

Regarding the additional config option I'm unsure. If people want CodiMD-content to be highly available, and they're using an image proxy, that proxy should be highly available too.

Reply:
Yes, let's not try to be smart here. If one configures an image proxy, use the proxy, if one doesn't then don't. That easy. :+1:

Reply:
Should be done now